universal-application-tool-0.0.1/app/assets/javascripts/admin_validation.ts

```diff
@@ -0,0 +1,74 @@
+/** The validation controller provides basic client-side validation of admin form fields. */
+class AdminValidationController {
+  static readonly MULTI_OPTION_QUESTION_FIELD_NAME_CREATE = '#question-settings input[name="newOptions[]"]';
+  static readonly MULTI_OPTION_QUESTION_FIELD_NAME_EDIT = '#question-settings input[name="options[]"]';
+
+  isMultiOptionCreateValid = true;
+  isMultiOptionEditValid = true;
+
+  constructor() {
+    // Attach listener to admin program edit form.
+    const adminProgramEditForm = document.getElementById("question-form");
+    if (adminProgramEditForm) {
+      adminProgramEditForm.addEventListener("submit", (event) => { return this.attemptSubmit(event); });
+    }
+  }
+
+  private attemptSubmit(event: Event): boolean {
+    this.checkFields();
+    if (!this.isValid()) {
+      event.preventDefault();
+      return false;
+    }
+    return true;
+  }
+
+  private checkFields() {
+    this.isMultiOptionCreateValid = this.validateMultiOptionQuestionCreate();
+    this.isMultiOptionEditValid = this.validateMultiOptionQuestionEdit();
+  }
+
+  updateFieldErrorState(element: Element, fieldName: string, isValid: boolean) {

```

Comment:
suggestion (blocking): it looks like this and the methods below are all for class-internal use only. please make them private as well for principle of least access

Reply:
Fixed.